### PR TITLE
Rework our netstandard1.3 test references

### PR DESCRIPTION
**Pull request by jaredpar**
The `netstandard1.3` test references were being provided by our
MS.CA.Test.Resources.Proprietary package. The contents of this package
get included in the output directory of every single unit test DLL that
we build (which contributes to build output size). The references
themselves though were only used in a very small number of tests.

This change upgrades to a version of MS.CA.Test.Resources.Proprietary
that has the `netstandard1.3` references removed and adds the
`netstandard1.3` references to only the projects that needs them

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler , @dotnet/roslyn-infrastructure PTAL

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -11530,6 +11531,165 @@ at Microsoft.CodeAnalysis.Diagnostics.AnalyzerManager.AnalyzerExecutionContext.<
         }
 #endif
 
+        private static ImmutableArray<byte> CreateCSharpAnalyzerNetStandard13(string analyzerAssemblyName)
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:11534]: note: this is not new code, I just moved it from the core test utilities project to here.

File: eng/Versions.props [hunk 2]
```diff
@@ -50,9 +50,11 @@
   -->
   <PropertyGroup>
     <BasicUndoVersion>0.9.3</BasicUndoVersion>
-    <BasicReferenceAssembliesNetStandard20Version>1.2.1</BasicReferenceAssembliesNetStandard20Version>
-    <BasicReferenceAssembliesNet50Version>1.2.1</BasicReferenceAssembliesNet50Version>
-    <BasicReferenceAssembliesNet60Version>1.2.2</BasicReferenceAssembliesNet60Version>
+    <BasicReferenceAssembliesNetStandard20Version>1.2.4</BasicReferenceAssembliesNetStandard20Version>
+    <BasicReferenceAssembliesNet50Version>1.2.4</BasicReferenceAssembliesNet50Version>
+    <BasicReferenceAssembliesNet60Version>1.2.4</BasicReferenceAssembliesNet60Version>
+    <BasicReferenceAssembliesNetStandard13>1.2.4</BasicReferenceAssembliesNetStandard13>
```
Comment (JoeRobich) [inline eng/Versions.props:56]: Appears to be unused.
```suggestion
```

**Review by JoeRobich — APPROVED** [review]

File: eng/Versions.props (hunk 2 above)
Comment (jaredpar) [reply eng/Versions.props:56]: I am bad at coding ...

**Pull request merged**